opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/ReferralLibrary.java

```diff
@@ -0,0 +1,173 @@
+
+    private ECSyncHelper syncHelper;
+
+    private ClientProcessorForJava clientProcessorForJava;
+    private Compressor compressor;
+
+    private ReferralLibrary(Context contextArg, Repository repositoryArg, int applicationVersion, int databaseVersion) {
+        this.context = contextArg;
+        this.repository = repositoryArg;
+        this.applicationVersion = applicationVersion;
+        this.databaseVersion = databaseVersion;
+    }
+
+    public static void init(Context context, Repository repository, int applicationVersion, int databaseVersion) {
+        if (instance == null) {
+            instance = new ReferralLibrary(context, repository, applicationVersion, databaseVersion);
+        }
+    }
+
+    public static ReferralLibrary getInstance() {
+        if (instance == null) {
+            throw new IllegalStateException(" Instance does not exist!!! Call "
+                    + CoreLibrary.class.getName()
+                    + ".init method in the onCreate method of "
+                    + "your Application class ");
+        }
+        return instance;
+    }
+
+    /**
+     * Use this method when testing.
+     * It should replace org.smartregister.Context#setInstance(org.smartregister.Context, org.smartregister.repository.Repository) which has been removed
+     *
+     * @param context
+     */
+    public static void reset(Context context, Repository repository, int applicationVersion, int databaseVersion) {
+        if (context != null) {
+            instance = new ReferralLibrary(context, repository, applicationVersion, databaseVersion);
+        }
+    }
+
+    public Context context() {
+        return context;
+    }
+
+    public Repository getRepository() {
+        return repository;
+    }
+
+    public int getApplicationVersion() {
+        return applicationVersion;
+    }
+
+    public int getDatabaseVersion() {
+        return databaseVersion;
+    }
+
+    public ECSyncHelper getEcSyncHelper() {
+        if (syncHelper == null) {
+            syncHelper = ECSyncHelper.getInstance(context().applicationContext());
+        }
+        return syncHelper;
+    }
+
+    public ClientProcessorForJava getClientProcessorForJava() {
+        if (clientProcessorForJava == null) {
+            clientProcessorForJava = ClientProcessorForJava.getInstance(context().applicationContext());
+        }
+        return clientProcessorForJava;
+    }
+
+    public void setClientProcessorForJava(ClientProcessorForJava clientProcessorForJava) {
+        this.clientProcessorForJava = clientProcessorForJava;
+    }
+
+    public Compressor getCompressor() {
+        if (compressor == null) {
+            compressor = Compressor.getDefault(context().applicationContext());
+        }
+        return compressor;
+    }
+
+    /**
+     * Use this method when testing purposes only.
+     * It should be replaced by data synchronized from the server
+     */
+    public void seedSampleReferralServicesAndIndicators() {
```

Comment:
@cozej4 I guess you added this for the purpose of testing along and will be removed when the library is shipped as a snapshot

Reply:
Yeah, this is for testing purposes only. It should be removed later on once we have the saver referral module running and synchronizing the services to the client app. The library will be shipped as a snapshot